src/WebJobs.Extensions.DurableTask/IPlatformInformationService.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Interface for accessing the AppService plan information,
+    /// the OS, and user-facing PL.
+    ///
+    /// Note: The functionality is currently limited, but will grow
+    /// along with the pursuit of more platform-specific defaults.
+    /// </summary>
+    public interface IPlatformInformationService
```

Comment:
Part of me wants to leave an [Obsolete] attribute on this, with an explanation that this is meant for internal purposes only and may change in the near future. Considering we are going to be adding methods at the very least, if not fully redesigning this interface in a subsequent release, I don't want customers to take a dependency on this via dependency injection.

Even once this class is finalized, I'm not convinced we want customers to inject their own value for this ever?

Reply:
I agree. I actually completely forgot that this was public. Happy to add `[Obsolete]` as the design is still WIP and also we never want customers to add DI here.

Would making this `internal` help prevent customer DI?

Reply:
It would, but there are issues making these internal, because the constructor of DurableTaskExtension has to be public I believe (not sure if this is a Functions extension limitation or a DI restriction). And you can't put an internal type on a public constructor :(.